### PR TITLE
chore: fix template build warnings

### PR DESCRIPTION
**Pull request by edbzn**

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "sourceMap": true,
     "declaration": false,
     "moduleResolution": "node",
-    "emitDecoratorMetadata": true,
```
Comment (hoebbelsB) [inline tsconfig.base.json:8]: isn't emitDecoratorMetadata: true angular default setting? Why do we turn it off?

Comment (edbzn) [reply tsconfig.base.json]: I removed this change, the correct way to fix these warnings 

```
WARNING: "ChangeDetectorRef", "NgZone", "TemplateRef" and "ViewContainerRef" are imported from external module "@angular/core" but never used in "dist/libs/template/esm2015/if/lib/if.directive.js" and "dist/libs/template/esm2015/if/lib/if.module.js".
```

is to upgrade to ng-packagr.

Comment (hoebbelsB) [reply tsconfig.base.json]: I don't get it. Do we still have the warning now? Can we remove it by upgrading ng-packagr? If so, why didn't u? :D

Comment (edbzn) [reply tsconfig.base.json]: @hoebbelsB This PR fixes multiple warnings, but not all, some of them are related to ng-packagr, and we have to update the whole @nrwl/workspace to update ng-packagr, so I'll make another PR for the workspace. Last time I tried I had different issues, so I'll need to work on it further. This PR can be merged as it.

**Pull request merged**